src/main/groovy/nebula/plugin/dependencylock/tasks/DiffLockTask.groovy

```diff
@@ -0,0 +1,151 @@
+
+        contents.each { configuration, dependencyLockMap ->
+            Map<String, String> dependencies = lock.get(configuration)
+            dependencyLockMap.each { dependency, fieldMap ->
+                dependencies.put(dependency, fieldMap.get("locked"))
+            }
+        }
+
+        return lock
+    }
+
+    Map<String, List<String>> performDiff(Map<String, Map<String, String>> old, Map<String, Map<String, String>> updated) {
+        Map<String, SortedMap<String, Tuple2<String, String>>> diffByConfiguration = collectConfigurationDiffs(old, updated)
+
+        Map<String, Map<Tuple2<String, String>, List<String>>> collapsedDiffs = [:].withDefault { [:].withDefault { [] } }
+        diffByConfiguration.forEach { configuration, configurationDiff ->
+            configurationDiff.forEach { dependency, versions ->
+                Map<Tuple2<String, String>, List<String>> versionDiffs = collapsedDiffs.get(dependency)
+                versionDiffs.get(versions).add(configuration)
+                collapsedDiffs.put(dependency, versionDiffs)
+            }
+        }
+
+        LinkedHashMap<String, List<String>> diff = filterDependencies(collapsedDiffs)
+
+        return diff
+    }
+
+    Map<String, List<String>> filterDependencies(Map<String, Map<Tuple2<String, String>, List<String>>> collapsedDiffs) {
+        SortedMap<String, Tuple2<String, String>> consistent = collapsedDiffs.findAll { it.value.size() == 1 }.collectEntries { k, v -> [(k): v.keySet()[0]] }.sort()
+
+        List<String> newDependencies = consistent.findAll { it.value.first == '' }
+                .collect { dependency, version -> "$dependency: ${version.getSecond()}".toString() }
+        List<String> removedDependencies = consistent.findAll { it.value.second == '' }
+                .collect { dependency, _ -> dependency }
+        List<String> updatedDependencies = consistent.findAll { it.value.first != '' && it.value.second != '' }
+                .collect { dependency, versions -> "$dependency: ${versions.getFirst()} -> ${versions.getSecond()}".toString() }
+
+        SortedMap<String, Map<Tuple2<String, String>, List<String>>> inconsistent = collapsedDiffs.findAll { it.value.size() > 1 }.sort()
+        List<String> inconsistentDependencies = inconsistent.inject([]) { collector, dependency, versionConfigurations ->
+            collector.add("$dependency:")
+            versionConfigurations.forEach { versions, configurations ->
+                collector.add("  ${versions.getFirst()} -> ${versions.getSecond()} [${configurations.sort().join(',')}]")
+            }
+            collector
+        }
+
+        return ['new': newDependencies, 'removed': removedDependencies, 'updated': updatedDependencies, 'inconsistent': inconsistentDependencies]
+    }
+
+    Map<String, SortedMap<String, Tuple2<String, String>>> collectConfigurationDiffs(Map<String, Map<String, String>> old, Map<String, Map<String, String>> updated) {
```

Comment:
Do you think we should use some explicit types?

Reply:
some types added, and methods simplified